internal/configs/hcl2shim/mock_value_composer.go

```diff
@@ -119,8 +118,8 @@ func (mvc MockValueComposer) composeMockValueForAttributes(schema *configschema.
 
 			for _, err := range typeConformanceErrs {
 				diags = diags.Append(tfdiags.WholeContainingBody(
-					tfdiags.Warning,
-					fmt.Sprintf("Ignored mock/override field `%v`", k),
+					tfdiags.Error,
+					fmt.Sprintf("Invalid mock/override field `%v`", k),
 					fmt.Sprintf("Values provided for override / mock must match resource fields types: %v.", err),
 				))
 			}
```

Comment:
Since those checks are errors now, we don't want to proceed with processing. Let's put a continue after this loop.